### PR TITLE
Sites Management Page: Use `<DropdownMenu />` core component

### LINKED ISSUE #66725: Sites Management Page: Use standard WP style for drop down
This isn't a blocker for i1.

Take a look at the ellipsis menu for each site in the Automattic P2 dashboard: cMuAv-p2. It follows the standard WordPress component style
Image

Whereas the SMP is currently using the Calypso style drop down
Image

@SaxonF mentioned to me that the P2 dashboard is an example of what the our dropdown in the SMP should look like.

Plus it will ultimately make the table component more portable down the track.

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #66725

## Proposed Changes

Uses the `<DropdownMenu />` core component instead of Calypso's `<EllipsisMenu />`.

The icon looks a little different than the original issue because P2 uses its own ellipsis icon and I opted to save bundle size by reusing an existing `<Gridicon />`.

### After

image

### Before

image

## Testing Instructions

1. Navigate to `/sites-dashboard`.
2. Verify the dropdown menu appears and works as expected.

### CODE REVIEW

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const ManagePluginsItem = ( { site, recordTracks }: SitesMenuItemProps ) => {
 		: [ getPluginsUrl( site.slug ), __( 'Plugins' ) ];
 
 	return (
-		<PopoverMenuItem
+		<MenuItem
 			href={ href }
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx:79]: This `href` is throwing a TypeScript warning and I don't know why 😦

Comment (sejas) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx:79]: - @danielbachhuber, here is a way to "fix" it https://github.com/Automattic/wp-calypso/pull/66879

Ideally, we should use another component that supports `href` or we should change the types on https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__components/button/index.d.ts#L134

Comment (danielbachhuber) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx:79]: Fixed with #66879

Comment (danielbachhuber) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx:79]: ... and thanks for your help, @sejas !

**Review by vykes-mac — COMMENTED** [review]
Seems like the favoured component to use. One thing I don't like though is that the highlighter remained on the first menu item even when I hover other items, I think the highlighter should move to the item I hover. 

https://user-images.githubusercontent.com/47489215/186218555-76213a95-0eff-47ea-b6b4-1777d90f3534.mov

**Comment by danielbachhuber** [comment]
> One thing I don't like though is that the highlighter remained on the first menu item even when I hover other items, I think the highlighter should move to the item I hover.

@vykes-mac Looks like there's an existing issue on the topic! https://github.com/WordPress/gutenberg/issues/26401

It seems like there should be a blue hover state. Here's the Block Editor:

https://user-images.githubusercontent.com/36432/186220269-f5e769b3-48b6-4e81-94f7-97b09845f4c4.mp4

It seems like we're missing the `--color-neutral-70` CSS variable for some reason...

image

**Comment by vykes-mac** [comment]
I see from the existing issue seems like the blue highlighter is for focus purposes only and the blue text is for hover. It would seem the `--color-neutral-70` is what's preventing the blue text when we hover. remove it and the blue text shows up.

**Comment by zaguiini** [comment]
The table is wider now that we're using a different component.

Adding

```css
inset-inline-end: 18px;
position: relative;
```

to the dropdown (`.components-dropdown components-dropdown-menu`) element on the table row fixes it.

Also, adding `display: flex` to the grid item version, along with

```css
padding-block: 0;
height: auto;
```

on the button (`.components-button.has-icon`) restores the same line height for the title element for every grid item. It occupies more vertical space now due to the size of the button.

**Comment by danielbachhuber** [comment]
> The table is wider now that we're using a different component.

@zaguiini Oh, good eye. I missed that before.

I've added a couple of tweaks with 989e79150df245b29e3e93981b7d93597b4c31d3 and f116d3aa03513d8d12c28ca875cf3b9dff6adf3b

Here's how it looks now:

image

image


> I see from the existing issue seems like the blue highlighter is for focus purposes only and the blue text is for hover. It would seem the `--color-neutral-70` is what's preventing the blue text when we hover. remove it and the blue text shows up.

@p-jackson Any good ideas on how to sort out this color mess?

**Review by zaguiini — APPROVED** [review]

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx (hunk 1 above)
Comment (p-jackson) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx:79]: Yes there's definitely just a type error here. Tracing through the Gutenberg code I found `<MenuItem>` is just a wrapper around `<Button>` which explicitly accepts an `href` prop.

**Comment by p-jackson** [comment]
> Any good ideas on how to sort out this color mess?

Yeah it is a bit of a mess 😮‍💨 

Core button styles are defined here: https://github.com/WordPress/gutenberg/blob/8b9db3dc025f73e9dad9b076c6f14394b898a9b5/packages/components/src/button/style.scss#L19-L24
They're using the `--wp-admin-theme-color` variable for the hover colour.

Over in Calypso we override the core styles in `_wp-components-overrides.scss`: https://github.com/Automattic/wp-calypso/blob/1455eefcb7c34091911f13f7f85a63f9b7e36372/client/assets/stylesheets/_wp-components-overrides.scss#L15-L18

That's where `--color-neutral-70` comes from. It resets all button colours back to "neutral" before applying our Calypso variables on top. Presumably so it matches the user's chosen UI theme.

[If it's a "primary" button it applies one set of styles](https://github.com/Automattic/wp-calypso/blob/1455eefcb7c34091911f13f7f85a63f9b7e36372/client/assets/stylesheets/_wp-components-overrides.scss#L29-L31). If it's a "secondary" button it applies a different set of styles. Same thing for "tertiary".

But then that's it, it doesn't apply any other styles, so any other button variant doesn't get styled.

I thought I'd just have to add a new override for whatever variant type the menu item is, but alas `<MenuItem>` has _no_ variant (it's not primary, tertiary, link, or anything). So I might have to do something hackier.

Does @Automattic/team-calypso-components have any advice?

**Comment by p-jackson** [comment]
I've just pushed an override that fixes hover state just for our menu items.

But now that I think of it, maybe we should add the styling for `.components-menu-item__button` to `_wp-components-overrides.scss` 🤔

**Comment by ciampo** [comment]
I'm not really familiar with all the style overrides in Calypso, but are all of them necessary at all? Can't `Button` be used as-is?

Ideally, we should avoid overriding component styles as much as possible. Also, those classnames are not guaranteed to be public APIs, and relying on them is definitely prone to style breakage over time.

We're soon going to the experiment with adding some (simple) theming options to components, which hopefully will remove the need for the majority of the style overrides

**Comment by p-jackson** [comment]
> which hopefully will remove the need for the majority of the style overrides

That sounds great! My guess is the overrides are needed because the core components don't use enough variables.

For example, rather than overriding the `color: var(--wp-admin-theme-color);` rule that `<Button>` uses on hover, we could instead redefine the `--wp-admin-theme-color` variable to match the user's theme preference. However, [the non-hover colour doesn't use a CSS variable](https://github.com/WordPress/gutenberg/blob/8b9db3dc025f73e9dad9b076c6f14394b898a9b5/packages/components/src/button/style.scss#L19), and so we can't tweak it the same way. But maybe that just needs a PR to core.

**Comment by ciampo** [comment]
> But maybe that just needs a PR to core.

Definitely! Feel free to open a PR (or even just an issue) in core and tag @mirka and myself :)

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 2]
```diff
@@ -66,44 +84,59 @@ const ManagePluginsItem = ( { site, recordTracks }: SitesMenuItemProps ) => {
...
+	'> div[role="group"]': {
+		display: 'flex',
+		flexDirection: 'column',
+	},
+} );
+
+const SiteDropdownMenu = styled( DropdownMenu )( {
+	'> .components-button': {
+		padding: '0px',
+		minWidth: '0px',
```
Comment (zaguiini) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]:
```suggestion
		padding: 0,
		minWidth: 0,
```

Let's remove the units here: https://stackoverflow.com/questions/7923042/when-specifying-a-0-value-in-css-should-i-explicitly-mark-the-units-or-omit

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 3]
```diff
@@ -66,44 +84,59 @@ const ManagePluginsItem = ( { site, recordTracks }: SitesMenuItemProps ) => {
...
 };
 
-const alignExternalIcon = css`
-	.gridicons-external {
-		top: 0px;
-	}
-`;
+const ExternalGridIcon = styled( Gridicon )( {
+	top: '-1px',
+	marginLeft: '4px',
```
Comment (zaguiini) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]: Let's go with logical properties:

```suggestion
	insetBlockStart: '-1px',
	marginInlineStart: '4px',
```

**Review by zaguiini — APPROVED** [review]
A few changes but overall good to go!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7493951

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7493951).

**Pull request merged**